tools/pre-commit-hook.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+#
+# Pre-commit linting hook.
+#
+# Install from root of repository with:
+#
+#   cp tools/pre-commit-hook.py .git/hooks/pre-commit
+
+import subprocess
+import sys
+import os
+
+# names of files that were staged
+# add  '*.pxd', '*.pxi' once cython-lint supports it
+p = subprocess.run(['git', 'diff',
+                    '--cached', '--name-only', '-z',
+                    '--diff-filter=ACMR',
+                    '--', '*.py', '*.pyx'],
+                   capture_output=True, check=True)
+
+files = p.stdout.decode(sys.getfilesystemencoding()).split('\0')
+files = [f for f in files if f]
+
+# create a temporary copy of what would get committed, without unstaged
+# modifications (e.g., only certain changes in a file may have been committed)
+git_dir = os.environ.get('GIT_DIR', '.git')
```

Comment:
Can we put instead this? Not sure though but works for me.

But from the doc `GIT_DIR` is not available all the time.

> Before Git invokes a hook, it changes its working directory to either $GIT_DIR in a bare repository or the root of the working tree in a non-bare repository. An exception are hooks triggered during a push (pre-receive, update, post-receive, post-update, push-to-checkout) which are always executed in $GIT_DIR.


```suggestion
pwd = os.environ.get('PWD')
git_dir = os.environ.get('GIT_DIR', f'{pwd}/.git')
```

Reply:
os.cwd gives that too. Just depends if the hook is always run from repo root (I think so).

Reply:
I don't see how `{pwd}/.git` is different from `.git` here?

Reply:
The difference is that it will be an absolute path.

Reply:
So, are you thinking the hook will be called from anywhere other than the root of the repo?

Reply:
It can if you use an external GUI tool yes.